serverless/serverless
# Fixed #4188 - Package generating incorrect package artifact path in serverless-state.json

## What did you implement:

Closes #4188 - Package generating incorrect package artifact path in serverless-state.json

## How did you implement it:

Because artifacts are generated in the `.serverless` directory first, even if a package path was provided, the reference to the artefact zip file was incorrectly pointing at the `.serverless` directory.
This only needs to be changed at the time the service state is saved, so I have added a few lines to the `saveServiceState.js` file of the AWS package plugin to fix the paths in the service configuration before saving.

## How can we verify it:

* In your `serverless.yml` file, specify a package path like so:
```
package:
  path: ../dist
```
* Run the `sls package` command
* Verify that the `serverless-state.json` file in the `dist` directory references the `dist` directory for the package path of each function

## Todos:


***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO
